source/read/retrieve.txt

```diff
@@ -0,0 +1,163 @@
+.. _c-retrieve:
+
+=============
+Retrieve Data
+=============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: code examples, read, search, cursor
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-short+} to retrieve data from a
+MongoDB collection by using read operations. You can call the ``mongoc_collection_find_with_opts()``
+function to retrieve documents that match a set of criteria specified in a query filter.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+.. _c-retrieve-find:
+
+Find Documents
+--------------
+
+The ``mongoc_collection_find_with_opts()`` function retrieves documents from a collection. This
+function takes a **query filter** and returns all matching documents. A query filter is a
```

Comment:
S: These two sentences can probably be combined since they're saying similar things
```suggestion
The ``mongoc_collection_find_with_opts()`` function takes a **query filter** and returns all matching documents from a collection. A query filter is a
```